doc/py_tutorials/py_setup/py_setup_in_ubuntu/py_setup_in_ubuntu.markdown

```diff
@@ -0,0 +1,144 @@
+nothing complicated.<br>
+
+First we will install some dependencies. Some are compulsory, some are optional. Optional
+dependencies, you can leave if you don't want.<br>
+
+### Required Build Dependencies
+
+We need **CMake** to configure the installation, **GCC** for compilation, **Python-devel** and
+**Numpy** for creating Python extensions etc.<br>
+
+`sudo apt-get install cmake`<br>
+`sudo apt-get install python-devel numpy`<br>
+`sudo apt-get install gcc gcc-c++`<br>
+
+Next we need **GTK** support for GUI features, Camera support (libdc1394, libv4l), Media Support
+(ffmpeg, gstreamer) etc.<br>
+
+`sudo apt-get install gtk2-devel`<br>
+`sudo apt-get install libdc1394-devel`<br>
+`sudo apt-get install libv4l-devel`<br>
+`sudo apt-get install ffmpeg-devel`<br>
+`sudo apt-get install gstreamer-plugins-base-devel`<br>
+
+### Optional Dependencies
+
+Above dependencies are sufficient to install OpenCV in your ubuntu machine. But depending upon your
+requirements, you may need some extra dependencies. A list of such optional dependencies are given
+below. You can either leave it or install it, your call :)
+
+OpenCV comes with supporting files for image formats like PNG, JPEG, JPEG2000, TIFF, WebP etc. But
+it may be a little old. If you want to get latest libraries, you can install development files for
+these formats.
+<br>
+`sudo apt-get install libpng-devel`<br>
+`sudo apt-get install libjpeg-turbo-devel`<br>
+`sudo apt-get install jasper-devel`<br>
+`sudo apt-get install openexr-devel`<br>
+`sudo apt-get install libtiff-devel`<br>
+`sudo apt-get install libwebp-devel`<br>
+
+Several OpenCV functions are parallelized with **Intel(R) Threading Building Blocks** (TBB). But if
+you want to enable it, you need to install TBB first. ( Also while configuring installation with
+CMake, don't forget to pass -D WITH_TBB=ON. More details below.)
+<br>
+` sudo apt-get install tbb-devel`<br>
+### Downloading OpenCV
+
+To download the latest source from OpenCV's [GitHub Repository](https://github.com/opencv/opencv). (If you want to contribute to OpenCV choose this. For that, you need to install **Git** first.
+<br>
+`sudo apt-get install git`<br>
+`git clone https://github.com/opencv/opencv.git`
+<br>
+It will create a folder opencv in current directory. The cloning may
+take some time depending upon your internet connection.
+
+Now open a terminal window and navigate to the downloaded OpenCV folder. Create a new build folder
+and navigate to it.
+<br>
+`mkdir build`<br>
+`cd build`
+<br>
+### Configuring and Installing
+
+Now we have installed all the required dependencies, let's install OpenCV. Installation has to be
+configured with CMake. It specifies which modules are to be installed, installation path, which
+additional libraries to be used, whether documentation and examples to be compiled etc. Below
+command is normally used for configuration (executed from build folder).
+<br>`CMAKE_CONFIGURATION_TYPES=Debug;Release`
+<br>
```

Comment:
We have misunderstanding here.

`CMAKE_CONFIGURATION_TYPES` is internal Cmake variable about supported configuration typed. And there is no `RELEASE`, but `Release` is here (for `CMAKE_BUILD_TYPE` value, which if "Release" by OpenCV's default - so it can be just omitted).

Reply:
I am seriously having issues with cmake variables, can you please let me know the exact line to include.

Reply:
Something like this.

```.diff
-Below command is normally used for configuration (executed from build folder).
+Below command is normally used for configuration with OpenCV's default parameters (executed from build folder):

$ cmake ../

OpenCV defaults assume "Release" build type and installation path is "/usr/local".
-... drop CMake variables description ...
+... add reference to tutorial with OpenCV C++ compilation for additional information about CMake options: https://docs.opencv.org/master/d7/d9f/tutorial_linux_install.html (please use doxygen internal link instead of online URL) ...

+... cmake output example about found/working Python support ...

 --   Python 2:
 --     Interpreter:                 /usr/bin/python2.7 (ver 2.7.6)
 --     Libraries:                   /usr/lib/x86_64-linux-gnu/libpython2.7.so (ver 2.7.6)
 --     numpy:                       /usr/lib/python2.7/dist-packages/numpy/core/include (ver 1.8.2)
 --     packages path:               lib/python2.7/dist-packages
 -- 
 --   Python 3:
 --     Interpreter:                 /usr/bin/python3.4 (ver 3.4.3)
 --     Libraries:                   /usr/lib/x86_64-linux-gnu/libpython3.4m.so (ver 3.4.3)
 --     numpy:                       /usr/lib/python3/dist-packages/numpy/core/include (ver 1.8.2)
 --     packages path:               lib/python3.4/dist-packages

-... Drop documentation build words ...
+Suggest to use online documentation instead on https://docs.opencv.org/master
````